detectors/aws/eks/detector.go

```diff
@@ -139,50 +139,34 @@ func (eksUtils eksDetectorUtils) fileExists(filename string) bool {
 	return err == nil && !info.IsDir()
 }
 
-// fetchString executes an HTTP request with a given HTTP Method and URL string.
-func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
-	request, err := http.NewRequest(httpMethod, URL, nil)
-	if err != nil {
-		return "", fmt.Errorf("failed to create new HTTP request with method=%s, URL=%s: %w", httpMethod, URL, err)
-	}
-
-	// Set HTTP request header with authentication credentials
-	authHeader, err := getK8sCredHeader()
+// fetchString executes an HTTP Kubernetes client request with a given HTTP Method and URL string.
+func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string, API string) (string, error) {
+	// Get cluster configuration
+	confs, err := getClusterConfig(URL)
```

Comment:
This should only be used from inside a k8s cluster, so we should be able to use the [`InClusterConfig`](https://github.com/kubernetes/client-go/blob/master/examples/in-cluster-client-configuration/main.go#L42) rather than making our own assumptions about the locations of things.  Similarly, we shouldn't need the base URL as an option here as that is discoverable by the client.